### PR TITLE
[Refactor] Extract save and load functions of database and table for code reuse

### PR DESCRIPTION
**Pull request by xiangguangyxg**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 25 (08.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|25|08.00%|[4688, 4694, 4695, 4696, 4697, 4698, 4699, 4700, 4704, 4710, 4711, 4720, 4728, 4740, 4741, 4742, 4743, 4744, 4746, 4752, 4753, 4754, 4760]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -4683,27 +4683,49 @@ public void addOrReplaceAutoIncrementIdByTableId(Long tableId, Long id) {
...
+        writer.writeJson(database.getTables().size());
+        List<Table> tables = database.getTables();
+        for (Table table : tables) {
+            saveTable(writer, table);
+        }
+    }
+
+    // For code reuse
+    public static int getTableSavedJsonNum() {
+        return 1;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4704]: how to understand this?

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -4683,27 +4683,49 @@ public void addOrReplaceAutoIncrementIdByTableId(Long tableId, Long id) {
...
+            throws IOException, SRMetaBlockException {
+        writer.writeJson(database);
+        writer.writeJson(database.getTables().size());
+        List<Table> tables = database.getTables();
+        for (Table table : tables) {
+            saveTable(writer, table);
+        }
+    }
+
+    // For code reuse
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4702]: comments like this won't provide any help

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 1 above)
Comment (xiangguangyxg) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4704]: Construct a SRMetaBlockWriter object must provide a number of json to save. If we provide saveTable(), we must provide getTableSavedJsonNum()

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 2 above)
Comment (xiangguangyxg) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4702]: just to remind others not to delete the function

**Pull request closed without merging**